### PR TITLE
Translate Plugin: Simplified Config. 

### PR DESCRIPTION
**Pull request by vishalboin**
### Description
Added functionality for multiple sources and multiple targets
 
### Issues Resolved
https://github.com/opensearch-project/data-prepper/issues/1914
 
### Check List
- [x] New functionality includes testing.
- [x] New functionality has been documented.
- [ ] New functionality has javadoc added
- [ ] Commits are signed with a real name per the DCO

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/data-prepper/blob/main/CONTRIBUTING.md).

### CODE REVIEW

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java [hunk 1]
```diff
@@ -0,0 +1,204 @@
...
+    @AssertTrue(message = "The pattern values do not match the target type provided")
+    public boolean isPatternTypeValid() {
+        if (Objects.isNull(regexParameterConfig) || Objects.isNull(regexParameterConfig.getPatterns())) {
+            return true;
+        }
+        Map<String, Object> patterns = regexParameterConfig.getPatterns();
+        return patterns.keySet().stream().allMatch(key -> checkTargetValueType(patterns.get(key)));
+    }
+
+    private boolean checkTargetValueType(Object val) throws NumberFormatException {
```
Comment (kkondaka) [inline data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java:122]: This is a config file. Looks like a lot of these functions do not belong here. Convention is to put only config related functions in the `XXXConfig.java` file

Comment (vishalboin) [reply data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java:122]: I will move the helper functions to a separate class

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java [hunk 2]
```diff
@@ -0,0 +1,137 @@
...
+
+public class TargetsParameterConfig {
+    private final TypeConverter converter;
+    private final LinkedHashMap<Range<Float>, Object> rangeMappings = new LinkedHashMap<>();
+    private final Map<String, Object> individualMappings = new HashMap<>();
+    private final Map<Pattern, Object> compiledPatterns = new HashMap<>();
+    @JsonProperty("target")
+    @NotNull
+    @NotEmpty
+    private String target;
```
Comment (graytaylor0) [inline data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java:27]: If this is under `targets`, why not just name this `key`? `key` is a more accurate description of what is being configured here

**Review by graytaylor0 — APPROVED** [review]

File: data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java (hunk 2 above)
Comment (vishalboin) [reply data-prepper-plugins/translate-processor/src/main/java/org/opensearch/dataprepper/plugins/processor/translate/TargetsParameterConfig.java:27]: Yes there could be change in the naming for `target` option and `target_type` option. target can be changed to `key` or `name` and `target_type` will be re-named to `type`

**Review by kkondaka — APPROVED** [review]

**Pull request merged**